packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/common.rs

```diff
@@ -215,12 +215,24 @@ pub trait ComponentNode: ComponentState {
 ///
 /// Two behaviors can be automatically derived by the `ComponentChildren` macro
 /// based on helper attributes applied to the struct.
+/// (Note: additional options will be added later)
 /// - `#[pass_through_children]`: all children are passed through as the rendered children (default if no attributes)
 /// - `#[no_rendered_children]`: no children are passed to the renderer
 #[enum_dispatch]
 pub trait ComponentChildren {
-    /// Return the children that will be used in the flat dast sent to the renderer.
-    fn render_children(&self) -> &Vec<UntaggedContent>;
+    /// Keeping this obsolete function so that can see how old code worked for reference.
+    /// TODO: delete once not needed anymore
+    fn render_children_obsolete(&self) -> &Vec<UntaggedContent>;
+
+    /// Given the vector of all flat dast children,
+    /// return the vector of flat dast children that should be sent to the renderer.
+    ///
+    /// TODO: make this more efficient to not create children in the first place
+    /// in the case of `no_rendered_children`?
+    fn filter_rendered_children(
+        &self,
+        children: Vec<FlatDastElementContent>,
```

Comment:
I think this function should be passed a `QueryObject` that has a reference to the structure graph and the current component. You can then ask to iterate over your own children. But if you don't ask, no computation is done.

Reply:
I'm glad you worked this out on your branch!